lighthouse-core/lib/tracehouse/trace-processor.js

```diff
@@ -611,15 +611,22 @@ class TraceProcessor {
       });
     const frameIdToRootFrameId = this.resolveRootFrames(frames);
 
-    // Filter to just events matching the frame ID, just to make sure.
+    // Filter to just events matching the main frame ID, just to make sure.
     const frameEvents = keyEvents.filter(e => e.args.frame === mainFrameIds.frameId);
 
     // Filter to just events matching the main frame ID or any child frame IDs.
-    const frameTreeEvents = keyEvents.filter(e => {
-      return e.args &&
-        e.args.frame &&
-        frameIdToRootFrameId.get(e.args.frame) === mainFrameIds.frameId;
-    });
+    // In practice, there should always be FrameCommittedInBrowser events to define the frame tree.
+    // Unfortunately, many test traces do not include FrameCommittedInBrowser events due to minification.
+    // This ensures there is always a minimal frame tree and events so those tests don't fail.
+    let frameTreeEvents = [];
+    if (!frameIdToRootFrameId.has(mainFrameIds.frameId)) {
+      frameIdToRootFrameId.set(mainFrameIds.frameId, mainFrameIds.frameId);
+      frameTreeEvents = frameEvents;
```

Comment:
hm that's weird codecov, based on the comment I would expect this to have happened in our test traces 🤔 

@adamraine any ideas why it thinks this isn't covered?

Reply:
Hmmmmmm, I put a `throw Error()` in here and a bunch of unit tests failed. Seems like we got this covered.

Reply:
Looks like codecov magically switched to 100%